core/src/main/java/org/mskcc/cbio/portal/dao/DaoMutation.java

```diff
@@ -101,6 +101,7 @@ public static int addMutationEvent(ExtendedMutation.MutationEvent event) throws
         MySQLbulkLoader.getMySQLbulkLoader("mutation_event").insertRecord(
                 Long.toString(event.getMutationEventId()),
                 Long.toString(event.getGene().getEntrezGeneId()),
+                Integer.toString(DaoReferenceGenome.getInstance().getReferenceGenomeByName(event.getNcbiBuild())),
```

Comment:
can we please cache this? There could be many, many thousands of such records, all executing the same query here. I expect this will significantly slow down the importing....

Reply:
for sure. Let's have one!

Reply:
@pieterlukasse , I've added the cache system to DaoReferenceGenome class.

Reply:
nice, thanks!